### ISSUE TITLE
Error when querying fields with cypher directive and relationship directive that resolves an interface (at the same time) 

### ISSUE THREAD

**Issue by AccsoSG** [neo4j/graphql#2261]
**Describe the bug**
Querying a field that is resolved by a cypher directive and at the same time a field that represents a relationship to an interface returns an error.

**Type definitions**
````graphql
interface Product {
  id: ID! @id
  uri: String!
}

type ProgrammeItem implements Product {
  id: ID! @id
  uri: String! @cypher(statement: "RETURN 'example://programme-item/' + this.id")
  editions: [Edition!]! @relationship(type: "HAS_EDITION", direction: OUT)
}

type Edition {
  id: ID! @id
  uri: String! @cypher(statement: "RETURN 'example://edition/' + this.id")
  product: Product! @relationship(type: "HAS_EDITION", direction: IN)
}
````

**To Reproduce**

1. Run a server with the following code...
2. Execute the following Mutation...
3. Then run the following Query...
Here an error happens:
````graphql
query {
  editions {
    id
    uri
    product {
      __typename
    }
  }
}
````

These queries work:
````graphql
query {
  editions {
    id
    uri
  }
}
````
````graphql
query {
  editions {
    id
    product {
      __typename
    }
  }
}
````
````json
{
  "errors": [
    {
      "message": "Variable `this_uri` not defined (line 14, column 25 (offset: 422))\r\n\"RETURN this { .id, uri: this_uri, product: this_product } as this\"\r\n                         ^",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "editions"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "code": "Neo.ClientError.Statement.SyntaxError",
          "name": "Neo4jError"
        }
      }
    }
  ],
  "data": null
}
````

**System (please complete the following information):**
* Version: 3.10.1

**Additional context**
This bug does not exist with version 3.5.1.

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @AccsoSG. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by neo4j-team-graphql** [comment]
We've been able to confirm this bug using the steps to reproduce that you provided - many thanks @AccsoSG! :pray: We will now prioritise the bug and address it appropriately.

**Comment by angrykoala** [comment]
Hi @AccsoSG Thanks for this report, and sorry for the regression affecting you. This will be solved by #2266 and published  shortly
